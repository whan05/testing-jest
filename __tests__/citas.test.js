import expect from "expect";
import Citas from "../js/classes/Citas";

describe('Probar clases de citas', () => { 

    const citas =  new Citas()

    const id = Date.now()

    test("Agregar nuevas citas" , () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Whan',
            telefono: '86809168',
            fecha: '20-01-2023',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj)

        expect(citas).toMatchSnapshot()
    })

    test('actualizar cita', () => { 
        const citaActualizada = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Whan',
            telefono: '86809168',
            fecha: '20-01-2023',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada)

        expect(citas).toMatchSnapshot()
     })

     test('Eliminar cita', () => { 
        citas.eliminarCita(id)

        expect(citas).toMatchSnapshot()
      })
    
 })