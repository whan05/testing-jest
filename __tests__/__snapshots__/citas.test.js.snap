// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar clases de citas Agregar nuevas citas 1`] = `
Citas {
  "citas": [
    {
      "fecha": "20-01-2023",
      "hora": "10:30",
      "id": 1674473891858,
      "mascota": "Hook",
      "propietario": "Whan",
      "sintomas": "Solo duerme",
      "telefono": "86809168",
    },
  ],
}
`;

exports[`Probar clases de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar clases de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "20-01-2023",
      "hora": "10:30",
      "id": 1674473891858,
      "mascota": "Nuevo Nombre",
      "propietario": "Whan",
      "sintomas": "Solo duerme",
      "telefono": "86809168",
    },
  ],
}
`;
